### PR TITLE
Add check for failed monitoring device

### PR DESCRIPTION
**Pull request by prgmitchell**
### Description

Add a check to see if the audio monitoring device has failed to load.
### Motivation and Context

It is a very common issue in the support channels where a user is unable to hear their monitored audio because the device has failed to load. This added check and solution instructions should hopefully allow users to resolve this on their own without having to seek additional support.
### How Has This Been Tested?

Ran the log analyzer code locally with my changes, tested logs with audio monitor devices failing to make sure it displayed the message when applicable.

https://obsproject.com/logs/P37v0V9MO8QF8Ldk

firefox_K6vuwAPqbS

### Types of changes

- New feature (non-breaking change which adds functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation

### CODE REVIEW

**Comment by flaeri** [comment]
LGTM. Ran thru a battery of logs, and all seems well.

The only thing I think could be improved is the message given to the user:
```
To correct this, head to Settings>Audio in OBS and change your monitoring device. 
After restarting OBS, select the monitoring device you'd like to use.
```
It might seem a bit confusing. Change the monitoring device how? To what?
Saying "after restarting" when the user was never asked to restart.

I know the issue is that their existing monitoring device is invalidated, and im not entirely sure what the audio monitoring selection drop down will look like. From what I recall, you need to select *something else*, and then re-select the device you want. Maybe one could say "pick default", and then reselect the device you want (if it is not the default).

Its a tricky one, as you wont to be concise, but at the same time precise enough to where most users get it. I'll suggest:
```
To correct this, head to Settings>Audio in OBS and change your monitoring device to "Default".
If you do not with to use the "default" device, you can change it to the correct one after hitting "Apply".
```
I don't believe the restart is strictly required, but I could be wrong. My English is not great, so feel free to make it flow a bit better, should you chose to use that style.

**Comment by prgmitchell** [comment]
> 

I was also a bit uncertain with the message after reading it back a few times. The unfortunate thing is that for many, a restart is required for whatever reason. The bot command instructions for this issue from the Discord are as follows:

> 1) Go to Settings > Audio and set your monitoring device to something other than what it is now. Press Apply.
2) Restart OBS.
3) Go to Settings > Audio and set your monitoring device to the correct one. Press Apply.

So maybe it should just mimic those?

**Comment by flaeri** [comment]
I trust you on this, as you have seen that issue more than most I'd think. Yeah, that seems fine to me, but probably good to make a bit more concise so it dosent take up so much space. IE, no need to repeat `Go to Settings > Audio and set your monitoring device`.

**Comment by prgmitchell** [comment]
Updated PR, changed wording on the error description to match the wording of the related OBS Bot command.

File: checks/audio.py [hunk 1]
```diff
@@ -17,6 +17,12 @@
     """, re.VERBOSE)
 
 
+def checkMonitoringDevice(lines):
+    if search('audio_monitor_init_wasapi: Failed', lines):
+        return [LEVEL_CRITICAL, "Audio Monitoring Device Failure",
+                "Your audio monitoring device is failing to load. To correct this, go to Settings > Audio and set your monitoring device to something other than what it is now. Press Apply and restart OBS. After restarting OBS, go to Settings>Audio and set your monitoring device to Default."]
```
Comment (WizardCM) [inline checks/audio.py:23]:
```suggestion
                "Your audio monitoring device is failing to load. To correct this, go to Settings > Audio and set your monitoring device to something other than what it is now. Press Apply and restart OBS. After restarting OBS, go back to Settings > Audio and set your monitoring device to Default."]
```

**Comment by prgmitchell** [comment]
@WizardCM updated the message using your suggestion.

File: checks/audio.py [hunk 2]
```diff
@@ -17,6 +17,12 @@
     """, re.VERBOSE)
 
 
+def checkMonitoringDevice(lines):
+    if search('audio_monitor_init_wasapi: Failed', lines):
+        return [LEVEL_CRITICAL, "Audio Monitoring Device Failure",
+                "Your audio monitoring device is failing to load. To correct this, go to Settings > Audio and set your monitoring device to something other than what it is now. Press Apply and restart OBS. After restarting OBS, go back to Settings > Audio and set your monitoring device to Default."]
```
Comment (RytoEX) [inline checks/audio.py:23]:
```suggestion
                "Your audio monitoring device failed to load. To correct this, go to Settings > Audio and set your monitoring device to something other than what it is now. Press Apply and restart OBS. After restarting OBS, go back to Settings > Audio and set your monitoring device to Default."]
```

**Review by RytoEX — CHANGES_REQUESTED** [review]

**Comment by RytoEX** [comment]
Also, remember that commit message bodies should use full sentences and punctuation.

**Comment by prgmitchell** [comment]
PR updated with the commit suggestion and corrected my commit message.

**Review by RytoEX — CHANGES_REQUESTED** [review]
The commit message body must be wrapped at 72 characters.

Additionally, this check as implemented is Windows-specific, and it may be worth noting that in the commit message.

> > 3. Go to Settings > Audio and set your monitoring device to the correct one. Press Apply.
>
> So maybe it should just mimic those?

The analyzer message seems to suggest something different. Was this idea dropped?

**Comment by prgmitchell** [comment]
@RytoEX I have updated to commit message to wrap and include that is is checking on Windows specifically. Yes, the analyzer message in the PR currently is different than what was originally used but I did decide to drop that idea that you quoted. I think the wording of the current message makes it clear to the user on what is happening with their monitoring device and what needs to be done to fix it.

**Comment by RytoEX** [comment]
I guess I'm mildly concerned because I do not think we follow "Default device" changes in Windows for monitoring devices - yet.

**Comment by prgmitchell** [comment]
@RytoEX this is really just meant as a tool for users to fix this issue on their own. It comes up in the support channels at least once a day and is a very easy solve on the users end. Ultimately I think this boils down to an issue with Windows reassigning device ID's which isn't fixable. and there isn't any clear indication in the UI that your monitoring device has failed. It is only logged.

**Comment by RytoEX** [comment]
> Ultimately I think this boils down to an issue with Windows reassigning device ID's which isn't fixable.

We have two open PRs on obs-studio to follow "Default device" changes on Windows, which would mitigate this problem for users who select "Default", but there is no telling when we'll get them merged.

**Comment by RytoEX** [comment]
Please fix the merge conflicts on this PR.

**Comment by prgmitchell** [comment]
@RytoEX merge conflicts fixed.

**Comment by prgmitchell** [comment]
Messaging has been updated to match the support bot command.

**Review by RytoEX — APPROVED** [review]
Looks good to me.

**Pull request merged**